analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Constraints/RandomnessConstraint.cs

```diff
@@ -0,0 +1,32 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Constraints;
+
+internal sealed class RandomnessConstraint : SymbolicConstraint
```

Comment:
```suggestion
internal sealed class CryptographicConstrain : CryptograhicallyStrongArray
```
This constraint is only for arrays correct (in my case it is)? If not I'd propose `CryptograhicallyStrong`.

Reply:
It's also going to be for `Span<Byte>` and `ReadonlySpan<Byte>`. See #7552.
`CryptographicConstraint `is too generic, there can be multiple cryptographic constraints.
Any other ideas?

Reply:
`CryptographicallySecureConstrain`?

Reply:
I don't like the idea of creating a new constraint for something, that we already have => rename it instead.

`ByteArrayConstraint` was a better name than `RandomnesConstraint`. If you want to extend it to spans and so on, you can make it a `ByteCollectionConstraint`.

The `Opposite` doesn't make sense in this constraint. Just return `null`. 

Then, you should rename `Modified` to `CryptographicallyStrong`

Reply:
@pavel-mikula-sonarsource We had this discussion on the opposite yesterday - what if you set a collection to be cryptographically strong and then the user overrides it with something strong later?
Can we set the constrain to null or remove it?

Reply:
You can do whatever - set a new constraint, or remove existing.

But for sure, your logic will not need `.Opposiste` functionality to do so. The opposite is used in branching conditions, where constraint is infered from the branching branch. That's practically impossible with this constriant, because `array.IsCryptographicallyStrong()` method can not exist.